lib/cocina/generator/schema.rb

```diff
@@ -47,10 +47,13 @@ def model_attributes
       end
 
       def types
-        type_properties_doc = schema_doc.properties['type']
-        return '' if type_properties_doc.nil? || type_properties_doc.enum.nil?
+        type_schema_property = schema_properties.find { |schema_property| schema_property.key == 'type' }
```

Comment:
This is necessary because `*WithMetadata` use `allOf`. This case wasn't previously handled for generating types.